lib/matplotlib/axes/_axes.py

```diff
@@ -4241,9 +4238,9 @@ def _parse_scatter_color_args(c, edgecolors, kwargs, xshape, yshape,
         else:
             try:  # First, does 'c' look suitable for value-mapping?
                 c_array = np.asanyarray(c, dtype=float)
-                n_elem = c_array.shape[0]
-                if c_array.shape in [xshape, yshape]:
-                    c = np.ma.ravel(c_array)
+                n_elem = c_array.size
+                if n_elem == xsize:
+                    c = c_array.ravel()
```

Comment:
did we need this cast to masked here?

Reply:
I don't think so.  The method will work regardless, since c_array is some kind of array, and I don't think we are later depending on c being a masked array.